src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -223,26 +223,12 @@ void Ekf::controlMagFusion()
 				const bool is_fusion_failing = isTimedOut(aid_src.time_last_fuse, _params.reset_timeout_max);
 
 				if (is_fusion_failing) {
-					if (_nb_mag_3d_reset_available > 0) {
-						// Data seems good, attempt a reset (mag states only unless mag_3D currently active)
+					if (!using_ne_aiding || !_control_status.flags.mag_aligned_in_flight) {
 						ECL_WARN("%s fusion failing, resetting", AID_SRC_NAME);
-						resetMagStates(_mag_lpf.getState(), _control_status.flags.mag_hdg || _control_status.flags.mag_3D);
+						resetMagStates(_mag_lpf.getState(), false);
 						aid_src.time_last_fuse = _time_delayed_us;
 
-						if (_control_status.flags.in_air) {
-							_nb_mag_3d_reset_available--;
-						}
-
-					} else if (starting_conditions_passing) {
-						// Data seems good, but previous reset did not fix the issue
-						// something else must be wrong, declare the sensor faulty and stop the fusion
-						//_control_status.flags.mag_fault = true;
-						ECL_WARN("stopping %s fusion, starting conditions failing", AID_SRC_NAME);
-						stopMagFusion();
-
 					} else {
-						// A reset did not fix the issue but all the starting checks are not passing
-						// This could be a temporary issue, stop the fusion without declaring the sensor faulty
 						ECL_WARN("stopping %s, fusion failing", AID_SRC_NAME);
 						stopMagFusion();
```

Comment:
The control logic can try to restart immediately, but it won't really restart because `fuseMag` in the starting process will return false due to the high test ratio.